### PR TITLE
chore: optimize local testing performance

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

Skip all assets build in local test environments:
- **BucketDeployment** asset build
- **NodejsFunction** code build
- **Backend** api docker build

**Local testing time(`pnpm test`): 1070s --> 180s**

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: test/control-plane/alb-control-plane-stack.test.ts [hunk 1]
```diff
@@ -22,6 +22,9 @@ import { SOLUTION_CONFIG_PATH } from '../../src/control-plane/private/solution-c
 import { TestApp, removeFolder } from '../common/jest';
 import { validateSubnetsRule } from '../rules';
 
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+jest.mock('aws-cdk-lib/aws-lambda-nodejs', () => require('../cdk-lambda-nodejs-mock'));
```
Comment (zxkane) [inline test/control-plane/alb-control-plane-stack.test.ts:26]: In the build history of Action, the `build` workflow still cost the same time as previous.

Does it work in ci environment?

Comment (tyyzqmf) [reply test/control-plane/alb-control-plane-stack.test.ts:26]: Use **CI** to determine if it is in the testing environment.

File: src/common/s3-asset.ts [hunk 2]
```diff
@@ -88,6 +83,19 @@ export function uploadBuiltInJarsAndRemoteFiles(
   };
 }
 
+function getSourceAsset(props: BuildJarProps, bundling: BundlingOptions) {
+  if (process.env.CI === 'true') {
```
Comment (zxkane) [inline src/common/s3-asset.ts:87]: You should change to `false`. We must do full validation in CI. You might ignore some code built in local development.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: test/analytics/modeling-on-athena/data-modeliing-athena-stack.test.ts [hunk 3]
```diff
@@ -15,6 +15,9 @@ import { App } from 'aws-cdk-lib';
 import { Match, Template } from 'aws-cdk-lib/assertions';
 import { DataModelingAthenaStack } from '../../../src/data-modeling-athena-stack';
 
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+jest.mock('aws-cdk-lib/aws-lambda-nodejs', () => require('../../cdk-lambda-nodejs-mock'));
```
Comment (zxkane) [inline test/analytics/modeling-on-athena/data-modeliing-athena-stack.test.ts:19]: Some here. Only mock it in local environments.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: test/jestEnv.js [hunk 4]
```diff
@@ -36,7 +36,7 @@ process.env.TEMPLATE_FILE = 'cloudfront-s3-control-plane-stack-global.template.j
 process.env.SOLUTION_VERSION = 'v1.1.5-202403071513'
 
 // web console bundling
-process.env.IS_SKIP_ASSET_BUNDLE = 'true'
+process.env.CI = 'true'
```
Comment (zxkane) [inline test/jestEnv.js:39]: remove it

**Review by zxkane — CHANGES_REQUESTED** [review]

**Review by zxkane — APPROVED** [review]

**Review by zxkane — APPROVED** [review]

**Review by dengmingtong — APPROVED** [review]

**Pull request merged**